services/webhook/discord.go

```diff
@@ -154,8 +154,14 @@ func (d discordConvertor) Push(p *api.PushPayload) (DiscordPayload, error) {
 	var text string
 	// for each commit, generate attachment text
 	for i, commit := range p.Commits {
-		text += fmt.Sprintf("[%s](%s) %s - %s", commit.ID[:7], commit.URL,
-			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
+		// limit the commit message display to just the summary, otherwise it would be hard to read
+		message := strings.Split(commit.Message, "\n")[0]
+
+		// a limit of 50 is set because GitHub does the same
+		if len(message) > 50 {
+			message = fmt.Sprintf("%.47s...", message)
```

Comment:
From reading the Go [docs](https://pkg.go.dev/fmt), it appears that this will treat `message` as bytes when truncating. Is there a better way of truncating the size of a Go string without having problems with unicode characters? I will continue to find a solution to this when I get the time.

Reply:
After further doc reading, I learned that precision is measured in runes when using the `%s` specifier (if I used `%x` it would be measured in bytes). So I think my current usage is fine.

Reply:
Are you able to test with unicode to see how discord would behave if we sent them truncated characters?

Reply:
I modified my test to contain unicode characters. I also realized `len(message)` returns the # of bytes in `message` rather than characters. Therefore, I opted to use `utf8.RuneCountInString()`.

However, it [appears](https://git-scm.com/docs/git-commit/2.14.6#_discussion) that git commit messages, although typically utf-8 encoded, could accept other encodings. git encourages use of utf-8 for commit messages, but it doesn't enforce it. So I am not sure if the way I count the # of characters in `message` and how I truncate `message` is correct. Using`utf8.RuneCountInString()` could come with a performance cost (i.e. if the commit summary is very large).